### PR TITLE
Change trading rebate rewards SOV become liquid

### PR DESCRIPTION
**Pull request by cwsnt**
Enable setter & getter for locked sov basis point
Added Deployment script & unit test for the setter & getter

### CODE REVIEW

File: contracts/core/State.sol [hunk 1]
```diff
@@ -204,6 +204,8 @@ contract State is Objects, ReentrancyGuard, Ownable {
 	mapping(address => mapping(address => uint256)) public specialRebates; // Special rate rebates for spesific pair -- if not set, then use the default one
 	bool public pause; //Flag to pause all protocol modules
 
+	uint256 internal tradingRebateRewardsBasisPoint;
```
Comment (ororopickpocket) [inline contracts/core/State.sol:207]: all variables which are added should be documented.
e.g.: 
Defines the portion of the trading fees which is to be paid out in a liquid form in basis points (0.01%). The rest is vested.

File: contracts/modules/ProtocolSettings.sol [hunk 2]
```diff
@@ -114,6 +116,15 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents, M
 		emit SetLockedSOVAddress(msg.sender, oldLockedSOVAddress, newLockedSOVAddress);
 	}
 
+	function setTradingRebateRewardsBasisPoint(uint256 newBasisPoint) external onlyOwner whenNotPaused {
```
Comment (ororopickpocket) [inline contracts/modules/ProtocolSettings.sol:119]: documentation missing.
never add functions without documentation.

File: contracts/modules/ProtocolSettings.sol [hunk 3]
```diff
@@ -651,4 +662,8 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents, M
 	function isProtocolPaused() external view returns (bool) {
 		return pause;
 	}
+
+	function getTradingRebateRewardsBasisPoint() external view returns (uint256) {
```
Comment (ororopickpocket) [inline contracts/modules/ProtocolSettings.sol:666]: same here

File: contracts/modules/ProtocolSettings.sol [hunk 4]
```diff
@@ -114,6 +116,15 @@ contract ProtocolSettings is State, ProtocolTokenUser, ProtocolSettingsEvents, M
 		emit SetLockedSOVAddress(msg.sender, oldLockedSOVAddress, newLockedSOVAddress);
 	}
 
+	function setTradingRebateRewardsBasisPoint(uint256 newBasisPoint) external onlyOwner whenNotPaused {
+		require(newBasisPoint <= 10000, "value too high");
```
Comment (ororopickpocket) [inline contracts/modules/ProtocolSettings.sol:120]: lockedSOV only allows 9999, does it not? So, we should have the same restriction here.

**Review by ororopickpocket — CHANGES_REQUESTED** [review]

**Review by eMarchenko — APPROVED** [review]

**Pull request merged**